### PR TITLE
[YouTube] Fix parsing short relative date formats (English only)

### LINKED ISSUE #1067: [YouTube] New short date format
YouTube is A/B testing a new, shorter format for video upload dates.

The long format is still used for comments and playlist update dates.

### en-US

- Year: `y`
- Month: `mo`
- Week: `w`
- Day: `d`
- Hour: `h`
- Minute: `min`

### en-GB

- Year: `yr`
- Month: `mo`
- Week: `wk`
- Day: `day(s)`
- Hour: `hr`
- Minute: `min`

image

The A/B test currently occurs very rarely (14 out of 1000 requests).

Here is a visitor data cookie for testing: `CgtJeFI1cWVSWVBTNCj6yc6jBg%3D%3D`

### PR DESCRIPTION
**Pull request by Theta-Dev**
- [X] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [X] no API changes

I added support for the new, short date format (e.g. 1wk ago).

Fixes #1067

### CODE REVIEW

**Comment by TobiGr** [comment]
LGTM, but what about the other languages? I'd guess we need to update them, too.

File: extractor/src/main/java/org/schabi/newpipe/extractor/localization/TimeAgoParser.java [hunk 1]
```diff
@@ -93,7 +93,7 @@ private boolean textualDateMatches(final String textualDate, final String agoPhr
         final String escapedSeparator = patternsHolder.wordSeparator().equals(" ")
                 // From JDK8 → \h - Treat horizontal spaces as a normal one
                 // (non-breaking space, thin space, etc.)
-                ? "[ \\t\\xA0\\u1680\\u180e\\u2000-\\u200a\\u202f\\u205f\\u3000]"
```
Comment (FireMasterK) [inline extractor/src/main/java/org/schabi/newpipe/extractor/localization/TimeAgoParser.java:96]: Why is `\d` added here? Could you update the above comment to indicate this?

**Review by FireMasterK — APPROVED** [review]

File: extractor/src/main/java/org/schabi/newpipe/extractor/localization/TimeAgoParser.java (hunk 1 above)
Comment (Theta-Dev) [reply extractor/src/main/java/org/schabi/newpipe/extractor/localization/TimeAgoParser.java:96]: To be able to parse strings where numbers directly follow digit (eg. `1wk`. I'll add that to the comment

**Comment by Theta-Dev** [comment]
Parsing YouTube data in other languages is currently disabled and it would require more extensive changes to the dictionary and the parser.

I have implemented (and tested) a parser that works with all languages. Here is the dictionary for it. Also note that there are some special cases that have to be handled seperately (e.g. in French `a` is both an article and the short form of "years").

https://code.thetadev.de/ThetaDev/rustypipe/src/branch/main/testfiles/dict/dictionary.json

**Review by TobiGr — APPROVED** [review]

**Comment by AudricV** [comment]
> Parsing YouTube data in other languages is currently disabled

Yes, but the timeago parser can be also used for something else than YouTube by clients, even if that's its main goal.

Is the data you provided extracted from YouTube? Do you wish to do other languages support using your similar approach to us? I think we should have a parser separating date units (seconds, hours, days, ...), digits (1, 2, 3, ...) and number units (tens, hundreds, thousands, ...) for each language we want to add support.

> in French a is both an article and the short form of "years"

Did you mean the verb `have` at the third singular person and the short form of `years`? There is not `a` article, but a `à` one.

**Comment by Theta-Dev** [comment]
This is the French term in question (5 years ago):

`il y a 5 a`

The data from the parsing dictionary is a combination of data extracted from YouTube and the CLDR repository.

**Comment by AudricV** [comment]
Merging this PR, as we would need to overhaul the timeago parser system to work with other languages on short time units. Thanks for the fix!

**Pull request merged**